contracts.py

```diff
@@ -34,6 +34,26 @@ class UnknownClassException(Exception):
 
 class BackendContractsPlugin(BasePlugin):
     def on_page_markdown(self, markdown, page, config, files, **kwargs):
+        """
+        Replace backend/contract markers with appropriate templates
+
+        on_page_* methods are called for each Page in a mkdocs site and can
+        modify the markdown they are given as input.  We're using this method
+        to look for the backend and contract markers, defined in the *_pat
+        variables above, and lookup the defined classes in the JSON file
+        generated by the databuilder.
+
+        For example:
+
+            !!! backend:cohortextractor2.concepts.Patients
```

Comment:
So the confusion I had before was where this lived; I hadn't twigged that the mkdocs site was the opensafely documentation.  I think that's probably obvious to anyone more familiar with mkdocs.  Maybe a small addition like:
```
on_page_* methods are called for each Page in a mkdocs site (in this case, the OpenSAFELY documentation site) and can....
```
Then again, this is a plugin and maybe we want to keep things more generic.

Reply:
Good point!  This is just a plugin to document our tooling in our platform docs so I'm happy for it to be really specific 👌